source/adapters/level_zero/v2/queue_immediate_in_order.cpp

```diff
@@ -10,8 +10,48 @@
 
 #include "queue_immediate_in_order.hpp"
 
+#include "../kernel.hpp"
+#include "../program.hpp"
+
 namespace v2 {
 
+ur_wait_list_t::ur_wait_list_t(const ur_event_handle_t *phWaitEvents,
```

Comment:
this seems overcomplicated.

Let's do something like:
```
class ur_wait_list {
std::pair<ze_event_handle_t *, size_t> getArray(ur_event_handle_t  *urHandles, size_t size, ze_event_handle_t extra) {
  size_t real_size = size + (extra ? 1 : 0);
  handles.reserve(real_size);
  for (size_t i = 0; i < size; ++i) {
     handles[i] = urHandles[i]->zeHandle;
  }
  
  if (extra) 
    handles[size] = extra;

  return (handles.data(), real_size );
}

private:
  std::vector<ze_event_handle_t *> handles; //preallocated, never shrinks
}
```

Reply:
Good idea. Done.